tests/test_messaging.py

```diff
@@ -1316,6 +1335,384 @@ def test_send_fcm_error_code(self, status):
         assert json.loads(recorder[0].body.decode()) == body
 
 
+class TestSendAll(object):
+
+    _PAYLOAD_FORMAT = """--boundary\r\nContent-Type: application/http\r\n\
+Content-ID: <uuid + 1>\r\n\r\nHTTP/1.1 {} Success\r\n\
+Content-Type: application/json; charset=UTF-8\r\n\r\n{}\r\n\r\n--boundary--"""
+    _CLIENT_VERSION = 'fire-admin-python/{0}'.format(firebase_admin.__version__)
+
+    @classmethod
+    def setup_class(cls):
+        cred = testutils.MockCredential()
+        firebase_admin.initialize_app(cred, {'projectId': 'explicit-project-id'})
+
+    @classmethod
+    def teardown_class(cls):
+        testutils.cleanup_apps()
+
+    def _instrument_batch_messaging_service(self, app=None, status=200, payload=''):
+        if not app:
+            app = firebase_admin.get_app()
+        fcm_service = messaging._get_messaging_service(app)
+        if status == 200:
+            content_type = 'multipart/mixed; boundary=boundary'
+        else:
+            content_type = 'application/json'
+        fcm_service._transport = HttpMockSequence([
+            ({'status': str(status), 'content-type': content_type}, payload),
+        ])
+        return fcm_service
+
+    def _get_url(self, project_id):
+        return messaging._MessagingService.FCM_URL.format(project_id)
+
+    def test_no_project_id(self):
+        def evaluate():
+            app = firebase_admin.initialize_app(testutils.MockCredential(), name='no_project_id')
+            with pytest.raises(ValueError):
+                messaging.send_all([messaging.Message(topic='foo')], app=app)
+        testutils.run_without_project_id(evaluate)
+
+    @pytest.mark.parametrize('msg', NON_LIST_ARGS)
+    def test_invalid_send_all(self, msg):
```

Comment:
Another test case for 100+ messages

Reply:
Should we add validation here in `send_all` or fallback on the API for validation?

Reply:
I added validation in `send_all` for now, since we already have validation making sure `messages` is a list. We can remove though if falling back to the API limits is better.

Reply:
Validating here is the right thing to do. We can save the user an RPC call by catching it here.